### PR TITLE
feat(resolver): Add label dependency type to new resolver

### PR DESCRIPTION
**Pull request by dinhxuanvu**
**Description of the change:**
Introduce the new label dependency as predicate for the resolver.
Also add new function to match label dependency against operator
properties.

Signed-off-by: Vu Dinh <vdinh@redhat.com>

**Motivation for the change:**
Add an option for operator authors to fine-tune the dependency selection in case of conflicts. The label dependency can be used as a mark/filter for the resolver to select the desirable operator during dependency resolution.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
/hold
Put a hold here pending further review and approval.

**Comment by ecordell** [comment]
looks good - can you add a test case to the `resolver_tests`?

**Comment by dinhxuanvu** [comment]
@anik120 @ecordell Test case added. PTAL

File: pkg/controller/registry/resolver/resolver_test.go [hunk 1]
```diff
@@ -419,6 +419,75 @@ func TestSolveOperators_WithGVKDependencies(t *testing.T) {
...
+	Provides := APISet
+
+	namespace := "olm"
+	catalog := registry.CatalogKey{"community", namespace}
+
+	newSub := newSub(namespace, "packageA", "alpha", catalog)
+	subs := []*v1alpha1.Subscription{newSub}
+
+	deps := []*api.Dependency{
+		{
```
Comment (benluddy) [inline pkg/controller/registry/resolver/resolver_test.go:433]: This test relies on many more behaviors than just satisfying label dependencies. How about two test cases: both with two operators (A and B) available, a subscription to A, and a label dependency for A. Then in the first test case, B has the label and both are installed, and in the second, B does not have the label and resolution fails.

**Pull request merged**